admin-ui/plugins/auth-server/redux/sagas/ScriptSaga.js

```diff
@@ -0,0 +1,77 @@
+import { call, all, put, fork, takeLatest, select } from 'redux-saga/effects'
+import { isFourZeroOneError } from 'Utils/TokenController'
+import {
+  getScriptsResponse,
+  getScriptsByTypeResponse,
+  getScriptResponse,
+} from '../features/scriptSlice'
+import { getAPIAccessToken } from 'Redux/features/authSlice'
+import ScriptApi from '../api/ScriptApi'
+import { getClient } from 'Redux/api/base'
+import * as JansConfigApi from 'jans_config_api'
+
+function* newFunction() {
+  const token = yield select((state) => state.authReducer.token.access_token)
+  const issuer = yield select((state) => state.authReducer.issuer)
+  const api = new JansConfigApi.CustomScriptsApi(getClient(JansConfigApi, token, issuer))
+  return new ScriptApi(api)
+}
+
+export function* getScripts({ payload }) {
+  try {
+    const api = yield* newFunction()
+    const data = yield call(api.getAllScripts, payload.action || {})
+    yield put(getScriptsResponse({ data }))
+  } catch (e) {
+    yield put(getScriptsResponse(null))
+    if (isFourZeroOneError(e)) {
+      const jwt = yield select((state) => state.authReducer.userinfo_jwt)
+      yield put(getAPIAccessToken(jwt))
+    }
+  }
+}
+
+export function* getScriptsByType({ payload }) {
+  try {
+    const api = yield* newFunction()
+    const scriptType = payload.action?.type || payload.action || 'PERSON_AUTHENTICATION'
+    const data = yield call(api.getScriptsByType, scriptType, payload.action || {})
+    yield put(getScriptsByTypeResponse({ data }))
```

Comment:
Bug: scriptType can become an object when options are passed.

If payload.action is an object of options, current code assigns it to scriptType. This will break the API call.

```diff
-    const scriptType = payload.action?.type || payload.action || 'PERSON_AUTHENTICATION'
-    const data = yield call(api.getScriptsByType, scriptType, payload.action || {})
+    const isString = typeof payload?.action === 'string'
+    const scriptType =
+      payload?.action?.type || (isString ? payload.action : 'PERSON_AUTHENTICATION')
+    const options =
+      !isString && typeof payload?.action === 'object' ? payload.action : {}
+    const data = yield call(api.getScriptsByType, scriptType, options)
```